packages/services/api/src/modules/audit-logs/providers/audit-logs-manager.ts

```diff
@@ -0,0 +1,326 @@
+      organization: organizationSlug,
+    });
+    if (!isOwner) {
+      return {
+        error: {
+          message: 'Unauthorized: You are not authorized to perform this action',
+        },
+        ok: null,
+      };
+    }
+
+    const formattedStartDate = filter?.startDate ? new Date(filter.startDate) : undefined;
+    const formattedEndDate = filter?.endDate ? new Date(filter.endDate) : undefined;
+
+    try {
+      const { email } = await this.session.getViewer();
+      const getAllAuditLogs = await this.getPaginatedAuditLogs(
+        organizationSlug,
+        { first: 1000, after: 0 },
+        { startDate: formattedStartDate, endDate: formattedEndDate },
+      );
+
+      if (!getAllAuditLogs || !getAllAuditLogs.data || getAllAuditLogs.data.length === 0) {
+        return {
+          ok: null,
+          error: {
+            message: 'No audit logs found for the given organization',
+          },
+        };
+      }
+
+      const data = getAllAuditLogs.data.map(log => {
+        const { id, timestamp, user_id, user_email, event_action, metadata } = log;
+        const newUser = {
+          id: metadata.user.id,
+          email: metadata.user.email,
+          provider: metadata.user.provider,
+          fullName: metadata.user.fullName,
+          displayName: metadata.user.displayName,
+        };
+        delete metadata.user;
+        const newMetadata = {
+          ...newUser,
+          ...metadata,
+        };
+        return {
+          ID: id,
+          Type: event_action,
+          CreatedAt: timestamp,
+          UserID: user_id,
+          UserEmail: user_email,
+          Metadata: JSON.stringify(newMetadata),
+        };
+      });
+
+      const columns = ['ID', 'Type', 'CreatedAt', 'UserID', 'UserEmail', 'Metadata'];
+      const currentDate = new Date().toLocaleDateString();
+      const currentTime = new Date().toLocaleTimeString();
+
+      const customHeader = [
+        [`Hive Audit Logs for Organization: ${organizationSlug}`],
+        [`Date: ${currentDate}`],
+        [`Time: ${currentTime}`],
+        [`User: ${email}`],
+        [''], // Blank row
+        [''], // Another blank row to separate sections
+      ];
+
+      const csvData = await new Promise<string>((resolve, reject) => {
+        stringify(
+          data,
+          {
+            header: true,
+            columns,
+          },
+          (err, output) => {
+            if (err) {
+              reject(err);
+            } else {
+              const csv = customHeader.join('\n') + '\n' + output;
+              resolve(csv);
+            }
+          },
+        );
+      });
+
+      const s3Storage = this.s3Config[0];
+      const { endpoint, bucket, client } = s3Storage;
+      const cleanStartDate = formattedStartDate?.toISOString().split('T')[0];
+      const cleanEndDate = formattedEndDate?.toISOString().split('T')[0];
+      const key = `audit-logs/${organizationSlug}/${cleanStartDate}-${cleanEndDate}.csv`;
```

Comment:
TODO @TuvalSimha Let's add a unique id to the file name to make sure we do not accidentially overwrite files.
```
const key = `audit-logs/${organizationSlug}/${cleanStartDate}-${cleanEndDate}-${randomHash}.csv`;
```

Reply:
randomHash should probably the current unix timestamp.

```
const key = `audit-logs/${organizationSlug}/${unixTimestampInSeconds}-${cleanStartDate}-${cleanEndDate}.csv`
```